templates/_helpers.tpl

```diff
@@ -91,7 +103,7 @@ Generate comma-separated list of nodes for MongoDB-HA connection string, based o
 {{- range $index0 := until $replicas -}}
   {{- $index1 := $index0 | add1 -}}
   {{- if eq $architecture "replicaset" }}
-    {{- $mongo_fullname }}-{{ $index0 }}.{{ $mongo_fullname }}-headless{{ if ne $index1 $replicas }},{{ end }}
+    {{- $mongo_fullname }}-{{ $index0 }}.{{ $mongo_fullname }}-headless.{{ $.Release.Namespace }}.svc.cluster.local{{ if ne $index1 $replicas }},{{ end }}
```

Comment:
This would break my setup because my clusters are not named "cluster.local". Have you tried using `dnsConfig.searches`?

Reply:
Hmm. I guess my setup does not actually use this section, so it would not actually break my setup. I don't use the mongo/redis subcharts. But if I did install them in my cluster, this would prevent resolution because `cluster.local` does not exist.

So, I still wonder if adjusting `dnsConfig` would fix the issue for you.

Reply:
isn't `svc.cluster.local` a constant in k8?
yes i tried `dnsConfig.searches` and it didn't work.

Reply:
i will add a values parameter for an optional cluster domain instead of 'cluster.local'